crates/pathfinder/src/state/sync/pending.rs

```diff
@@ -78,14 +78,17 @@ pub async fn poll_pending<S: GatewayApi + Clone + Send + 'static>(
                     prev_block = Some(block.clone());
                     tracing::trace!("Pending block data changed");
 
-                    download_classes_and_emit_event(
+                    if let Err(e) = download_classes_and_emit_event(
```

Comment:
This is not quite correct I guess. Technically it's also possible that sending the event failed and I don't think we'd like to ignore that?

Reply:
I moved this check down into the classes download method.